### PR TITLE
geo/geomfn: implement ST_RotateX

### LINKED ISSUE #60893: geo/geomfn: implement ST_RotateX({geometry,float8})
Implement `ST_RotateX` on arguments {geometry,float8}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_RotateX.html).

_Observers: Please react to this issue if you need this functionality._

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end (your exhaustive unit tests go the above mentioned packages!).
* Ensure the documentation is regenerated by calling `make buildshort`. You can also play with it by calling `./cockroach demo --empty` afterwards.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

**The following additional guidance has been issued on implementing this function:**

use the existing affine transformation

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2021-02-21T20:42:26Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by busiest-beaver**
This patch implements `ST_RotateX`.

Resolves cockroachdb/cockroach#60893.

Release justification: low-risk updates to new functionality
Release note (sql change): `ST_RotateX` function is now available.

### CODE REVIEW

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 1]
```diff
@@ -4650,6 +4650,34 @@ The matrix transformation will be applied as follows for each coordinate:
...
+				{"g", types.Geometry},
+				{"angle_radians", types.Float},
+			},
+			ReturnType: tree.FixedReturnType(types.Geometry),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				g := tree.MustBeDGeometry(args[0])
+				rotRadians := float64(tree.MustBeDFloat(args[1]))
+
+				ret, err := geomfn.RotateX(g.Geometry, rotRadians)
+				if errors.Is(err, geomfn.ErrPointOriginEmpty) {
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:4666]: can you rearrange this to be:

```
	if err != nil {
       if errors.Is(err, geomfn.ErrPointOriginEmpty) {
          return tree.DNull, nil
			}
			return nil, err
   }
```

normally we're used to seeing `if err != nil` straight up, so it's nicer to read this way

**Review by otan — COMMENTED** [review]
Thanks! just the one comment

File: pkg/sql/sem/builtins/geo_builtins.go (hunk 1 above)
Comment (alsohas) [reply pkg/sql/sem/builtins/geo_builtins.go:4666]: Would be acceptable to just omit
```					
if errors.Is(err, geomfn.ErrPointOriginEmpty) {
    return tree.DNull, nil
}
``` 
And keep  
```
if err != nil {
    return nil, err
}
```
I think for this functionality it isn't necessary to check if point origin is empty, now that I take a closer look. Correct me if I'm wrong.

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 2]
```diff
@@ -4650,6 +4650,31 @@ The matrix transformation will be applied as follows for each coordinate:
...
+			},
+			ReturnType: tree.FixedReturnType(types.Geometry),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				g := tree.MustBeDGeometry(args[0])
+				rotRadians := float64(tree.MustBeDFloat(args[1]))
+
+				ret, err := geomfn.RotateX(g.Geometry, rotRadians)
+				if err != nil {
+					return nil, err
+				}
```
Comment (alsohas) [inline pkg/sql/sem/builtins/geo_builtins.go:4668]: Removed the unnecessary point origin is empty check from previous commit.

File: pkg/sql/sem/builtins/geo_builtins.go (hunk 1 above)
Comment (alsohas) [reply pkg/sql/sem/builtins/geo_builtins.go]: Just realized the point origin empty check is not needed actually, sorry about that. Should be okay now https://github.com/cockroachdb/cockroach/pull/61326#discussion_r585891554

**Review by otan — APPROVED** [review]
bors r+

**Pull request merged**